lib/base/utility.cpp

```diff
@@ -1788,7 +1788,7 @@ String Utility::ValidateUTF8(const String& input)
 		output.insert(output.end(), (const char*)l_Utf8Replacement, (const char*)l_Utf8Replacement + 3);
 	}
 
-	return String(output.begin(), output.end());
```

Comment:
... therefore, by moving the string here, this keeps that likely wasted memory allocated whereas the old code copied again into a probably more appropriately sized buffer.

So in the current state, this PR probably leads to some increase of memory usage. I'd suggest reserving only the length of the original string, assuming it's probably fine. If it's not, it might be reallocated once.